tsdb/engine/tsm1/cache.go

```diff
@@ -112,11 +113,15 @@ type Cache struct {
 
 // NewCache returns an instance of a cache which will use a maximum of maxSize bytes of memory.
 // Only used for engine caches, never for snapshots
-func NewCache(maxSize uint64, path string) *Cache {
+func NewCache(maxSize uint64, config tsdb.ShardConfig) *Cache {
```

Comment:
The shard config contains 4 fields:

* `Path`, which was previously used in NewCache
* `WALPath`, which is ignored in NewCache
* `Database` and `RetentionPolicy`, which are now used as tags on the Cache (and Filestore and WAL) so that it's trivial to match up which Shards and which parts of the engine stats line up together